js-packages/@prestojs/rest/src/Endpoint.ts

```diff
@@ -527,45 +602,39 @@ export default class Endpoint<ReturnT = any> {
      * The [UrlPattern](doc:UrlPattern) this endpoint hits when executed.
      */
     urlPattern: UrlPattern;
-    private transformResponseBody?: (data: any) => any;
+    public requestInit: ExecuteInitOptions;
     private urlCache: Map<string, Map<{}, PreparedAction>>;
     private decodeBody: (res: Response) => any;
-    private requestInit: ExecuteInitOptions;
     private resolveUrl: (
         urlPattern: UrlPattern,
         urlArgs?: Record<string, any>,
         query?: Query
     ) => string;
-    private middleware: MiddlewareFunction<ReturnT>[];
+    private middleware: Middleware<ReturnT>[];
 
     /**
      * @param urlPattern The [UrlPattern](doc:UrlPattern) to use to resolve the URL for this endpoint
      */
-    constructor(urlPattern: UrlPattern, options: EndpointOptions = {}) {
+    constructor(urlPattern: UrlPattern, options: EndpointOptions<ReturnT> = {}) {
         const {
             decodeBody = defaultDecodeBody,
-            transformResponseBody,
             resolveUrl = defaultResolveUrl,
-            middleware = Endpoint.defaultConfig.middleware,
+            middleware = [],
+            getMiddleware = Endpoint.defaultConfig.getMiddleware,
             ...requestInit
         } = options;
+
         this.urlPattern = urlPattern;
-        this.transformResponseBody = transformResponseBody;
         this.urlCache = new Map();
         this.decodeBody = decodeBody;
         this.requestInit = requestInit;
         this.resolveUrl = resolveUrl;
-        this.middleware = middleware;
-    }
-
-    /**
-     * Get the Paginator class to use for this endpoint.
-     *
-     * `Endpoint` defaults to no paginator. See [PaginatedEndpoint](doc:PaginatedEndpoint)
-     * for class that provides default paginator.
-     */
-    getPaginatorClass(): PaginatorInterfaceClass<any> | null {
-        return null;
+        this.middleware = getMiddleware(middleware, this);
+        this.middleware.forEach(m => {
+            if (typeof m !== 'function' && m.contributeToClass) {
```

Comment:
contributeToClass seems even less apt now: it's not modifying a class, it's modifying an endpoint instance

Reply:
ok, I will change it to `init`